src/inmanta/plugins.py

```diff
@@ -626,19 +654,28 @@ def get_annotation(arg: str) -> object:
             self.kwargs[arg] = argument
             self.all_args[arg] = argument
 
-    def get_signature(self) -> str:
+    def get_signature(self, dsl_types: bool = False) -> str:
```

Comment:
```suggestion
    def get_signature(self, *, dsl_types: bool = False) -> str:
```
I prefer this signature (no pun intended), because it forces you to be explicit. `get_signature(True)` does not give any indication as to what it does without looking at the definition (in contrast to something like `add(1, 2)`). `get_signature(dsl_types=True)` is a lot more informative.

Reply:
Great point :+1:

Reply:
This applies to the other methods you added as well, but I won't block it for that.

Reply:
Refactored